ECMA2Yaml/ECMAHelper/Models/VersionedValue.cs

```diff
@@ -47,7 +48,7 @@ public VersionedValue(HashSet<string> monikers, T value)
             Value = value;
         }
     }
-
+    [Serializable]
```

Comment:
why is there so many `[Serializable]` attributes?

Reply:
it is used with the DeepClone method of Namespace class.

Reply:
I see, how about just make use of the [MemberwiseClone](https://docs.microsoft.com/en-us/dotnet/api/system.object.memberwiseclone?view=net-5.0#System_Object_MemberwiseClone) method on `Type`? For any `List` or `Dictionary` properties, we don't need to deep copy all the values, just cloning the collection itself will be enough. Something like:
```csharp
        public Type DeepCopy()
        {
            Type rval = (Type) MemberwiseClone();
            rval.BaseTypes = new List<BaseType>(BaseTypes);
            rval.InheritanceChains = new List<VersionedCollection<string>>(InheritanceChains);
            rval.InheritedMembers = new Dictionary<string, VersionedString>(InheritedMembers);
            // go on for all collection properties
            return rval;
        }
```